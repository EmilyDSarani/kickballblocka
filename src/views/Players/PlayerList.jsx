import React from 'react'
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getPlayers } from '../../services/players';

export default function PlayerList() {
    const [players, setPlayers] = useState([]);
    useEffect(()=>{
        getPlayers().then((sponse)=>setPlayers(sponse));
    })
    return (
        <>
        <div>
            <h1>Player Page</h1>
        </div>
        <ul>
            {players.map((player)=>{
                return(
                    <li key={player.id}>
                        <Link to={`/players/${player.id}`}>
                            {player.name}
                        </Link>
                    </li>
                );
            })}
        </ul>
        </>
    )
}
