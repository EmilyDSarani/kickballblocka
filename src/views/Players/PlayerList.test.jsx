import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import PlayerList from './PlayerList';

it('should render a player list', async () => {
    
    render(
    <MemoryRouter>
    <PlayerList />
    </MemoryRouter>);

    const player = await screen.findByText('Bud E. Guy', {exact: false})
    expect(player).toBeInTheDocument();
})

