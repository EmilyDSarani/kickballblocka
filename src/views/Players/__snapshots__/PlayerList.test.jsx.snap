// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a player list 1`] = `
<div>
  <div>
    <h1>
      Player Page
    </h1>
  </div>
  <ul />
</div>
`;
